client.go

```diff
@@ -131,55 +132,55 @@ func WithKeepAlives(keepAlives bool) ClientOption {
 	}
 }
 
-const federationTripperLifetime = time.Minute * 5 // how long to keep an entry
-const federationTripperReapInterval = time.Minute // how often to check for dead entries
+const destinationTripperLifetime = time.Minute * 5 // how long to keep an entry
+const destinationTripperReapInterval = time.Minute // how often to check for dead entries
 
 // nolint:maligned
-type federationTripper struct {
+type destinationTripper struct {
 	// transports maps an TLS server name with an HTTP transport.
-	transports      map[string]*federationTripperTransport
+	transports      map[string]*destinationTripperTransport
```

Comment:
Why do we do this? Why can't we share transports for different endpoints?

Reply:
Because we have to set `ServerName` in the `tls.Config` for the remote server SNI, so each destination needs its own `tls.Config`.